Back-end/src/main/java/com/bramgussekloo/projects/config/SecurityConfig.java

```diff
@@ -12,32 +11,22 @@
 import org.springframework.security.config.http.SessionCreationPolicy;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.security.crypto.password.PasswordEncoder;
+import com.bramgussekloo.projects.utils.GetPropertyValues;
 
 import javax.sql.DataSource;
 
 @Configuration
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
 
-    @Autowired
-    private DataSource dataSource;
+    private final DataSource dataSource;
 
-    public static String HashUserPassword(String password) {
-        return encoder().encode(password);
+    public SecurityConfig(DataSource dataSource) {
+        this.dataSource = dataSource;
     }
 
-    @Bean
-    public static javax.sql.DataSource createDataSource() {
-        String propFileName = "Database_config.properties";
-        String[] values = GetPropertyValues.getDatabasePropValues(propFileName);
-        org.postgresql.ds.PGPoolingDataSource ds = new org.postgresql.ds.PGPoolingDataSource();
-        ds.setDataSourceName("A Data Source");
-        ds.setServerName(values[3]);
-        ds.setDatabaseName(values[4]);
-        ds.setUser(values[1]);
-        ds.setPassword(values[2]);
-        ds.setPortNumber(5432);
-        return ds;
+    public static String HashUserPassword(String password) {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.